js/noms/src/path-test.js

```diff
@@ -277,8 +281,44 @@ suite('Path', () => {
     t('.foo[42]bar', 'Invalid operator: b');
     t('#foo', 'Invalid operator: #');
     t('!foo', 'Invalid operator: !');
-    t('@foo', 'Invalid operator: @');
-    t('@key', 'Invalid operator: @');
+    t('@foo', 'Unsupported annotation: @foo');
+    t('@key', 'Cannot use @key annotation at beginning of path');
     t('.foo[42]@soup', 'Unsupported annotation: @soup');
+    t('.foo@key', 'Cannot use @key annotation on: .foo');
+    t('.foo@key()', '@key annotation does not support arguments');
+    t('.foo@key(42)', '@key annotation does not support arguments');
+    t('.foo@type()', '@type annotation does not support arguments');
+    t('.foo@type(42)', '@type annotation does not support arguments');
+  });
+
+  test('type annotation', async () => {
+    const m = new Map([
+      ['string', 'foo'],
+      ['bool', false],
+      ['number', 42],
+      ['List<number|string>', new List([42, 'foo'])],
+      ['Map<bool, bool>', new Map([[true, false]])],
+    ]);
+
+    const s = newStruct('', {
+      str: 'foo',
+      num: 42,
+    });
+
+    const tests = [];
+
+    await m.forEach((v: string, k: Value) => {
+      tests.push(assertResolvesTo(getTypeOfValue(v), m, `["${k}"]@type`));
```

Comment:
I actually messed the type annotations up here. `v` is a `Value`, `k` is a `string`. It happened because the k/v arguments are the confusing way around, but the little refactor cleared things up.

Sad that Flow didn't figure this out... or maybe it *did* figure it out, the type annotations were wrong, and it just ignored them.